build/versionutils.py

```diff
@@ -0,0 +1,109 @@
+#!/bin/python3
+#
+# increment_version.py
+#
+# This source file is part of the FoundationDB open source project
+#
+# Copyright 2015-2025 Apple Inc. and the FoundationDB project authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import argparse
+import re
+import subprocess
+import sys
+
+
+VERSION_POSITIONS = ['MAJOR', 'MINOR', 'BUILD', 'PATCH']
+VERSION_LINE = re.compile(r'version\s*\=\s*(\d+)\.(\d+)\.(\d+)\.(\d+)')
+
+
+def incremented_version(version: (int, int, int, int), update_type: str) -> tuple[int, int, int, int]:
```

Comment:
This is more for my own efforts to learn python, but why is `version` `(int, int, int, int)`, but this returns a `tuple[int, int, int, int]`

Reply:
Hm, the official documentation say to style it like `tuple[int, int, int, int]`: https://docs.python.org/3/library/typing.html#annotating-tuples I guess `(int, int, int, int)` must be an alias. I've gone ahead and made it consistently go with `tuple[int, int, int, int]`